Implementing custom upload function

jerinjacob007 opened the pull request:
New configuration option 

`uploader : {
uploadByFile(file){
return ...
} `
similar to [image tool](https://github.com/editor-js/image#providing-custom-uploading-methods)

• Readme updated

### Solves
#5

talyguryn committed:
implement custom uploader

Update index.js

custom upload function

build

update response and loading preview

jerinjacob007 committed:
custom upload function

jerinjacob007 committed:
build

jerinjacob007 committed:
update response and loading preview

jerinjacob007 committed:
custom uploader

jerinjacob007 committed:
Update README.md

jerinjacob007 committed:
Update README.md

Update README.md

jerinjacob007 committed:
Update README.md

jerinjacob007 committed:
Merge branch 'master' of https://github.com/jerinjacob1999/attaches

jerinjacob007 committed:
Update uploader.js

neSpecc commented on README.md:
```diff
@@ -64,6 +64,7 @@ Attaches Tool supports these configuration parameters:
 | Field | Type     | Description        |
 | ----- | -------- | ------------------ |
 | endpoint | `string` | **Required:** endpoint for file uploading. |
+| uploader | `{uploadByFile: function}` |Optional custom uploading methods|
```
Please add more detailed description of this ability to the Readme

jerinjacob007 replied:
Ok I'll do that

neSpecc commented on src/uploader.js:
```diff
@@ -21,18 +21,52 @@ export default class Uploader {
+    upload.then((response) => {
       this.onUpload(response);
     }).catch((error) => {
+      console.warn(error);
```
do we need this log?

jerinjacob007 replied:
I can remove it

jerinjacob007 committed:
remove unwanted log

jerinjacob007 committed:
Update README.md

Added detailed custom uploading documentation
Removed required filed from endpoint

jerinjacob007 committed:
build

christoph-kluge commented:
Is this going to be merged anytime soon? What is misisng here?

neSpecc commented:
Please increment a minor version and we will be ready to merge

jerinjacob007 committed:
Update package.json

jerinjacob007 commented:
> Please increment a minor version and we will be ready to merge

Version changed to `1.1.2`

neSpecc commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@editorjs/attaches",
-  "version": "1.1.1",
+  "version": "1.1.2",
```
There is the new feature added, so `1.2.0` would be better

jerinjacob007 committed:
Update package.json

jerinjacob007 commented:
Version is `1.2.0` now
